index.js

```diff
@@ -4109,14 +4109,20 @@ export const trimComponents = (components, format) => {
   const keyCache = {};
   const filteredComponents = [];
   for (const comp of components) {
+    if (!comp.version)
```

Comment:
This change skips over packages without any version and removes them from the package list entirely. If those would be kept in the SBoM the `purl` would have an invalid syntax since the version would be missing.

Reply:
Version is optional in purl. https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#package-url-specification-v10x

We cannot ignore a package entirely if only the version is missing (like missing a nut or a gluten entirely in ingredients, which is more problematic).